waku/v2/waku_archive/driver/postgres_driver/postgres_driver.nim

```diff
@@ -0,0 +1,288 @@
+when (NimMajor, NimMinor) < (1, 4):
+  {.push raises: [Defect].}
+else:
+  {.push raises: [].}
+
+import
+  std/db_postgres,
+  std/strformat,
+  std/nre,
+  std/options,
+  std/strutils,
+  stew/[results,byteutils],
+  chronos
+
+import
+  ../../../waku_core,
+  ../../common,
+  ../../driver
+
+export postgres_driver
+
+type PostgresDriver* = ref object of ArchiveDriver
+  connection: DbConn
+  preparedInsert: SqlPrepared
+
+proc dropTableQuery(): string =
+  "DROP TABLE messages"
+
+proc createTableQuery(): string =
+  "CREATE TABLE IF NOT EXISTS messages (" &
+  " pubsubTopic VARCHAR NOT NULL," &
+  " contentTopic VARCHAR NOT NULL," &
+  " payload VARCHAR," &
+  " version INTEGER NOT NULL," &
+  " timestamp BIGINT NOT NULL," &
+  " id VARCHAR NOT NULL," &
+  " storedAt BIGINT NOT NULL," &
+  " CONSTRAINT messageIndex PRIMARY KEY (storedAt, id, pubsubTopic)" &
+  ");"
+
+proc insertRow(): string =
+ """INSERT INTO messages (id, storedAt, contentTopic, payload, pubsubTopic,
+  version, timestamp) VALUES ($1, $2, $3, $4, $5, $6, $7);"""
+
+proc new*(T: type PostgresDriver, storeMessageDbUrl: string): ArchiveDriverResult[T] =
+  var host: string
+  var user: string
+  var password: string
+  var dbName: string
+  var port: string
+  var connectionString: string
+  var dbConn: DbConn
+  try:
+    let regex = re("""^postgres:\/\/([^:]+):([^@]+)@([^:]+):(\d+)\/(.+)$""")
+    let matches = find(storeMessageDbUrl,regex).get.captures
+    user = matches[0]
+    password =  matches[1]
+    host = matches[2]
+    port = matches[3]
+    dbName = matches[4]
+    connectionString = "user={user} host={host} port={port} dbname={dbName} password={password}".fmt
+  except KeyError,InvalidUnicodeError, RegexInternalError, ValueError, StudyError, SyntaxError:
+    return err("could not parse postgres string")
+
+  try:
+    dbConn = open("","", "", connectionString)
+  except DbError:
+    return err("could not connect to postgres")
+
+  ok(PostgresDriver(connection: dbConn))
+
+method reset*(s: PostgresDriver): ArchiveDriverResult[void] {.base.} =
+  try:
+    let res = s.connection.tryExec(sql(dropTableQuery()))
+    if not res:
+      return err("failed to reset database")
+  except DbError:
+    return err("failed to reset database")
+  ok()
+
+method init*(s: PostgresDriver): ArchiveDriverResult[void] {.base.} =
+  try:
+    let res = s.connection.tryExec(sql(createTableQuery()))
+    if not res:
+      return err("failed to migrate database")
```

Comment:
would an error saying `failed to initialize` make more sense here? Or is this a prep for adding migrations in the future?